src/mumble/main.cpp

```diff
@@ -125,7 +125,26 @@ int main(int argc, char **argv) {
 	}
 #endif
 
-	Global::g_global_struct = new Global();
+	// This argument has to be parsed first, since it's value is needed to create the global struct,
+	// which other switches are modifying. If it is parsed first, the order of the arguments does not matter.
+	QStringList args = a.arguments();
+	if (args.contains(QLatin1String("-c")) || args.contains(QLatin1String("--config"))) {
+		int index = std::max(args.indexOf(QLatin1String("-c")), args.indexOf(QLatin1String("--config")));
```

Comment:
I wonder if we could minimize the lookups by doing something like
```
const int index1 = args.indexOf(QLatin1String("-c"));
const int index2 = args.indexOf(QLatin1String("--config"));

if (index1 >= 0 && index2 >= 0) {
	// error as it was specified twice
} else if (index1 >= 0 || index2 >= 0) {
	// Option is given once
	const int index = std::max(index1, index2);
}
```
This would also take care of the situation in which the user specified `-c` **and** `--config`. It's still not perfect though as it doesn't check whether either of the two was used multiple times and I wonder if it is actually worth checking for these silly cases in the first place :thinking: :shrug:

Reply:
Getting rid of some lookups would be nice. However I don't think it's necessary to handle multiple flags. Other programs (like bash for example) just take the last flag and ignore the rest.

But there is still room for improvement. I like this better:
```
const int index = std::max(args.indexOf(QLatin1String("-c")), args.indexOf(QLatin1String("--config")));
if (index >= 0) {
 	// do things
}
```

Reply:
With our approach we'd always use the first argument specified. I agree however that it is probably not necessary to handle this here :shrug: 

Yeah your code-suggestion looks good to me :+1: